<a href="https://colab.research.google.com/github/dishidhak/brown_dhakshin_demand_estimation/blob/main/demand_estimation_air_fryers_ipynb.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Demand Estimation and Market Analysis: Air Fryers
### Dishitha Dhakshin (qxk8wp), Emerson Brown (gsz5cv)
## 2. Demand Estimation

Welcome!

In [13]:
df = pd.read_csv("air_fryers_clean_brand_year.csv")

In [14]:
import statsmodels.api as sm
import pandas as pd

feature_cols = ['compact_share', 'dual_basket_share', 'oven_style_share',
                'rotisserie_share', 'window_share']

df_reg = pd.get_dummies(df, columns=['brand', 'year'], drop_first=True)

X_cols = ['avg_price', 'avg_rating'] + feature_cols + \
         [c for c in df_reg.columns if c.startswith('brand_') or c.startswith('year_')]

X = sm.add_constant(df_reg[X_cols].astype(float))
y = df_reg['log_brand_share'].astype(float)

model = sm.OLS(y, X).fit()

In [15]:
print("Estimated price coefficient:", model.params['avg_price'])

Estimated price coefficient: -0.026674298382822256


The estimated price coefficient is negative, meaning higher prices are associated with lower market share. This is important because it is consistent with the basic economic principle that as prices increase, demand for that product goes down. If the coefficient were positive, it would imply that higher prices increase demand, which is unrealistic for air fryers and would make the model unreliable.

In [16]:
model.params[feature_cols].sort_values(ascending=False)

,0
dual_basket_share,12.195470
compact_share,1.544834
rotisserie_share,1.332578
oven_style_share,-0.756775
window_share,-1.785580


Dual-basket, compact, and rotisserie features are associated with higher demand, as they have positive coefficients. In contrast, oven-style and window features have negative coefficients, indicating lower demand. This suggests that consumers tend to prefer more compact and multifunctional designs, while certain specialized features may be less desirable in this market.

In [17]:
model.params.filter(like='brand_').sort_values(ascending=False)

,0
brand_share,10.641167
brand_cuisinart,3.753607
brand_oster,2.219050
brand_nuwave,0.893572
brand_ninja,0.479746
brand_instant_pot,-0.180711
brand_ultrean,-0.498878
brand_dash,-0.841570
brand_gowise usa,-1.055355
brand_cosori,-1.171665


The brands with the largest positive dummy coefficients are Cuisinart and Oster, followed by NuWave and Ninja. These coefficients indicate that, relative to the omitted baseline brand, these brands have higher market share after controlling for price, ratings, and product features.

In contrast, brands such as Cosori, GoWISE USA, and Dash have negative coefficients, indicating lower market share relative to the baseline brand.

In [18]:
model.params.filter(like='year_').sort_values(ascending=False)

,0
year_2021,0.505247
year_2023,0.444882
year_2022,0.437888
year_2020,0.326385


The year with the largest dummy coefficient is 2021, followed by 2023 and 2022. These positive coefficients indicate that, relative to the omitted baseline year (2019), overall demand was higher in these years. This suggests that the air fryer market experienced increased demand after 2019, with 2021 showing the strongest increase.

In [19]:
print("R-squared:", model.rsquared)

R-squared: 0.8911255787819694


About 89.1% of the variation in log market share is explained by the model. This suggests a strong fit.